2015/presentations/Teoria Kolejek - matematyka i efektywne wytwarzanie oprogramowania - Wiktor Żołnowski

```diff
@@ -0,0 +1,110 @@
+
+## Podstawy Teorii Kolejek i Prawo Little’a
+Dlaczego akurat Teoria Kolejek? Coraz częściej mamy do czynienia z systemami przetwarzającymi duże ilości danych, komunikującyim się z innymi systemami, czy też po prostu działającymi asynchronicznie. W takich systemach stosuje się różnego typu kolejki (mniej lub bardziej bezpośrednio - w końcu komputery, procesory, RAM, czy dyski twarde też działają w oparciu kolejki). W takich systemach można, na podstawie danych, które zazwyczaj jesteśmy w stanie zebrać, oszacować na podstawie wzorów zaczerpniętych z Teorii Kolejek maksymalne obciążenie systemu, czy też policzyć czasy procesowania dużej ilości danych w zależności od ilości procesów lub wątków. Jest to temat bardzo szeroki i poświęcono mu już całe książki i prace naukowe. 
+
+W moim przypadku teoria ta znalazła zastosowanie do czegoś innego. Takim (nie)typowym systemem kolejkowym jest przecież zespół developerski do którego spływają wymagania. Jak by nie było wymagania te czy to w postaci karteczek na tablicy, stories w Jirze, czy tasków w Trello układają się w kolejki. 
+
+Pomocne w zrozumieniu tego dlaczego dany proces/system działa, a inny nie okazało się również Prawo Little'a mówiące o tym, że średnia liczba rzeczy/klientów w systemie jest równa iloczynowi średniego czasu przebywania w systemie oraz średniego tempa ich przybywania.
+
+##Dlaczego Scrum i Kanban (czy jakikolwiek inny proces wytwarzania oprogramowania) działają?
+
+Jako Agile Coach miałem już na koncie kilka udanych wdrożeń Scrum i Kanban w różnych zespołach. Oczywiście wiedziałem jak istotny w Scrum jest Sprint Planning oraz Backlog Refinement. Wiedziałem też, że kluczowym czynnikiem efektywnego zastosowania Kanban jest zastosowanie limitu Work in Process (WIP). Wiedza o tym, co jest kluczowe do działania danego procesu nie jest jednoznaczna z wiedzą na temat tego dlaczego akurat te, a nie inne czynniki są tak ważne. 
+
+Czy próbowaliście przekonać kiedyś, kogoś do stosowanie danej metody tylko poprzez stwierdzenie, że: “powinniśmy stosować tą metodę bo z mojego doświadczenia wynika, że ona zadziała”? Albo: “wszyscy tak robią i u nich działa, więc u nas też będzie - spróbujmy”? Czasami się udaje, ale częściej mamy do czynienia ze zdrowym sceptycyzmem. 
+
+Nasz ludzka natura często sprawia, że opieramy się temu, czego nie rozumiemy. Rzeczy niezrozumiałe wręcz wzbudzają w nas poczucie strachu przed nieznanym. Wystarczy spojrzeć na naszą mitologię i historię - jeśli nie wiemy jak i dlaczego coś działa, bardzo szybko przypisujemy temu nadprzyrodzone moce - potrzebujemy wyjaśnienia. 
+
+Prawo Little’a jest dobrym wyjaśnieniem tego dlaczego limit WIP i dobre planowanie w Sprincie są tak ważne. Ale o tym za chwilę.
+
+## Zastosowanie Teorii Kolejek - kilka bardziej przyziemnych przykładów
+
+Zanim przejdziemy do rzeczy, chciałbym abyście zastanowili się przez chwilę nad odpowiedzią na poniższe pytania dotyczące utylizacji zasobów.
+
+* Jak wygląda autostrada zutylizowana w 100%?
```

Comment:
proponuję użyć innego słowa niż "zutylizowana" bo kojarzy się to głównie z utylizacją odpadów i czytając to zdanie miałem przed oczyma jakąś ruinę autostrady :)